### PR TITLE
fix(azure): disable use-dns for secondary nics

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
fix(azure): disable use-dns for secondary nics

DNS resolution through secondary NICs is not supported on Azure. Disable
it.
    
Without this, we see seconds of delay resolving urls in cloud-init logs
from Jammy+, see SF ticket.
    
Per cjp256's comment, the first NIC under metadata.imds.network is ensured
to be the primary one. We use this to determine primary NICs instead of
relying on fragile driver and/or NIC names.
    
Fixes: SF #00380708
    
Co-authored-by: Calvin Mwadime <calvin.mwadime@canonical.com>
```

## Additional Context

Supersedes https://github.com/canonical/cloud-init/pull/5180, see comments for more context.
Closes:
- https://warthogs.atlassian.net/browse/CPC-4224
- https://warthogs.atlassian.net/browse/CPC-3999
- https://bootstack.canonical.com/cases/00380708

## Test Steps

```bash
make deb

export CLOUD_INIT_CLOUD_INIT_SOURCE=cloud-init_all.deb
export CLOUD_INIT_PLATFORM=azure

tox -e integration-tests -- tests/integration_tests/datasources/test_azure.py::test_azure_multi_nic_setup
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [x] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [x] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by aciba90** [comment]
@danpdraper and @cjp256, I have created this PR superseding #5180. Please review it, thanks.

**Review by cjp256 — APPROVED** [review]
LGTM, thanks!!  I'll be sure to verify when it lands in the daily.

File: tests/integration_tests/datasources/test_azure.py [hunk 1]
```diff
@@ -45,3 +48,70 @@ def test_azure_eject(session_cloud: IntegrationCloud):
...
+    nic_one = AzureCreateParams(f"ci-nic1-test-{us}", rg_params.name, None)
+    nic_two = AzureCreateParams(f"ci-nic2-test-{us}", rg_params.name, None)
+    with session_cloud.launch(
+        launch_kwargs={
+            "resource_group_params": rg_params,
+            "network_interfaces_params": [nic_one, nic_two],
+        }
+    ) as client:
+        _check_for_eject_errors(client)
+        if CURRENT_RELEASE.series == "bionic":
```
Comment (danpdraper) [inline tests/integration_tests/datasources/test_azure.py:100]: In the interest of consistency with the `skipif` decorator above, could you replace this predicate with `CURRENT_RELEASE == BIONIC`?

**Review by danpdraper — APPROVED** [review]

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Comment by cjp256** [comment]
Just wanted to say thanks again @aciba90! Testing looks great so far.

**Comment by aciba90** [comment]
Thank you, @cjp256, for the support and testing! Let us know if anything goes wrong.

**Pull request merged**